pkg/dns/service.go

```diff
@@ -314,6 +322,33 @@ func (s *Service) GetManagedZone(ctx context.Context, t traffic.Interface) (*v1a
 	return nil, fmt.Errorf("no managed zone found for traffic resource : %s", t.GetName())
 }
 
+// GetManagedZoneForDomain returns a ManagedZone for the given domain if one exists.
+// Currently, this returns the first matching ManagedZone found in the traffic resources own namespace, or if none is found,
+// it looks for the first matching one listed in the default ctrl namespace.
+func (s *Service) GetManagedZoneForDomain(ctx context.Context, domain string, t traffic.Interface) (*v1alpha1.ManagedZone, error) {
+	var managedZones v1alpha1.ManagedZoneList
+
+	if err := s.controlClient.List(ctx, &managedZones, client.InNamespace(t.GetNamespace()), client.MatchingFields{"spec.domainName": domain}); err != nil {
+		log.Log.Error(err, "unable to list managed zones in traffic resource NS")
```

Comment:
Should we be able to pull the controller-runtime log logger from the context here?

Reply:
Yes, I'll update this